File: atna_go/linkedlist.go
type Node struct {
	data interface{}
	nx   *Node
}

type mylist struct {
	head *Node
}

func (l *mylist) insert(val int) {
	nd := &Node{data: val} // make node
	p := l.head            //var for head

	if l.head == nil { // check if head is nil
		l.head = nd
	} else { // if not nil
		for p.nx != nil {
			p = p.nx
		}
		p.nx = nd
	}
}

func main() {
	kk := mylist{}
	kk.insert(1)
	kk.insert(2)
	kk.insert(3)
	f.Printf("%v", kk.head.data)
	f.Printf("%v", kk.head.nx.data)
	f.Printf("%v", kk.head.nx.nx.data)
}

File: atna_go/ply_grd_golang/dsa/linkedlist.py
class Node:
    def __init__(self,data):
        self.data = data
        self.nx = None

class mylist:
    def __init__(self):
        self.head = None

    def insert(self, val):
        nd = Node(val)
        p = self.head

        if self.head == None:
            self.head = nd
        else:
            while p.nx != None:
                p = p.nx
            p.nx = nd




if __name__ == "__main__":
    kk = mylist()
    kk.insert(1)
    kk.insert(2)
    kk.insert(3)
    print("{}".format(kk.head.data))
    print("{}".format(kk.head.nx.data))
    print("{}".format(kk.head.nx.nx.data))